src/java/org/apache/cassandra/index/sai/utils/IndexTermType.java

```diff
@@ -667,16 +672,31 @@ else if (!indexType.subTypes().isEmpty() && !indexType.isMultiCell())
             capabilities.add(Capability.VECTOR);
 
         if (indexType instanceof InetAddressType)
+        {
             capabilities.add(Capability.INET_ADDRESS);
+            reversed = false;
+        }
 
         if (indexType instanceof IntegerType)
+        {
             capabilities.add(Capability.BIG_INTEGER);
+            reversed = false;
+        }
 
         if (indexType instanceof DecimalType)
+        {
             capabilities.add(Capability.BIG_DECIMAL);
+            reversed = false;
+        }
 
         if (indexType instanceof LongType)
+        {
             capabilities.add(Capability.LONG);
+            reversed = false;
+        }
+
+        if (reversed)
+            capabilities.add(Capability.REVERSED);
```

Comment:
These will now have `isReversed()` return `false` during `Expression` creation to align w/ the fact we don't actually index reversed byte-comparables for these types.